packages/allow-scripts/test/index.js

```diff
@@ -61,17 +93,34 @@ test('cli - run command - good dep as a sub dep', async (t) => {
   let allowScriptsSrcRoot = path.join(__dirname, '..', 'src')
   let projectRoot = path.join(__dirname, 'projects', '3')
 
+  // clean up from a previous run
+  // the force option is only here to stop rm complaining if target is missing
+  fs.rmSync(path.join(projectRoot, './node_modules/bbb/.goodscriptworked'), { force: true })
+  fs.rmSync(path.join(projectRoot, './node_modules/bbb/node_modules/.bin'), {
+    recursive: true,
+    force: true
+  })
   // run the "run" command
   let cmd = path.join(allowScriptsSrcRoot, 'cli.js')
-  let result = spawnSync(cmd, ['run'], {cwd: projectRoot})
+  let result = spawnSync(cmd, ['run'], { cwd: projectRoot })
+
+  // forward error output for debugging
+  console.error(result.stderr.toString('utf-8'))
 
   // assert the output
   t.deepEqual(result.stdout.toString().split('\n'), [
+    'installing bin scripts',
+    '- bbb>good_dep',
     'running lifecycle scripts for event \"preinstall\"',
     '- bbb>good_dep',
     'running lifecycle scripts for event \"install\"',
     'running lifecycle scripts for event \"postinstall\"',
+    '- bbb',
     'running lifecycle scripts for top level package',
     '',
   ])
+
+  t.assert(fs.existsSync(path.join(projectRoot, './node_modules/bbb/node_modules/.bin/good')), 'Expected a nested bin script to be installed in bbb/node_modules/.bin')
+  t.assert(fs.existsSync(path.join(projectRoot, './node_modules/bbb/.goodscriptworked')), 'Expected a bin script to be available from a postinstall script')
```

Comment:
This test now also documents how bin scripts work - the nesting and the availability in PATH